### ISSUE TITLE
Create a shared conda environment to save time and to avoid potential disk quota issues on home dirs for some 

### ISSUE THREAD

**Issue by DininduSenanayake** [GenomicsAotearoa/Nextflow_Workshop#4]
@christopher-hakkaart I am going through the material now and took a note of few areas which we can save time and avoid potential issues with some of the regular users of the cluster
* Building the conda environment (even with `--solver=libmamba` takes about 4-5 minutes. 
* Final build adds up to about 4.8GB. Default location for conda envs is individual home dirs  which has a 20GB quota. Some regular users of the cluster have some usage of homedirs which will not provide the ~5GB buffer.In order to  avoid potential disk quota issues, It will be ideal to use a shared conda environment.  I have created one in `/nesi/project/nesi02659/.conda/envs/nf-core` which all of the workshop attendees have access to. Then I thought we can simplify the setup even further with a shared init script. Therefore, **Setup** instructions for both sessions were updated as below

**Session 1**

```bash
source /nesi/project/nesi02659/nextflow-workshop/init-nf-day1
```

**Session 2**

```bash
source /nesi/project/nesi02659/nextflow-workshop/init-nf-day2
```

and that's all as the init scripts do everything 

Also, I have removed the additional `module load Python/3.10.5-gimkl-2022a` and `module load Java/17` in favour of the python and java versions comes with the conda environment. Latter has py3.11 and java-20

**Comment by christopher-hakkaart** [comment]
Thanks @DininduSenanayake - makes sense. I didn't realize the spaces were limited and removing the modules is a great 👍 

Other formatting changes are great as well.

**Comment by DininduSenanayake** [comment]
Sorry for the last minute chane @christopher-hakkaart. I should have discussed this with you  weeks ago.  

>spaces were limited 

I am afraid so for the `/home` directories. There are terabytes of space in shared filesystems such as `/nesi/project/nesi02659` and `/nesi/nobackup/nesi02659` but `/home` is restricted to 20GB. ( when we ran the  first snakemake workshop where everyone had to create their own environments, we have re-directed everyone's working directories to `/nesi/project`. 

Overall, I think it is okay/safer  to provide the pre-built environment to everyone. Plus it takes away that 5 minutes of silence while everyone attempt to install the packages 😊 ( also what we did with the second snakeamake workshop. Move everyone to modules and a pre-built shared environment )

**Comment by DininduSenanayake** [comment]
Ah..I am adding `Java/17` back to the init script. Otherwise, runner fails with

```bash
$ NXF_VER=22.10.0 nextflow run nextflow-io/hello -r v1.1
N E X T F L O W  ~  version 22.10.0
NOTE: Your local project version looks outdated - a different revision is available in the remote repository [3b355db864]
Launching `https://github.com/nextflow-io/hello` [goofy_kalman] DSL1 - revision: baba3959d7 [v1.1]
BUG! exception in phase 'semantic analysis' in source unit 'Script_085f23c1' Unsupported class file major version 64
```

Confirming `module load Java/17` fixes the problem and yes it is


```bash
$ module load Java/17

The following have been reloaded with a version change:
  1) Java/11.0.4 => Java/17

(nf-core) 11:32:56 wbn004 ~/session1 $ NXF_VER=22.10.0 nextflow run nextflow-io/hello -r v1.1
N E X T F L O W  ~  version 22.10.0
NOTE: Your local project version looks outdated - a different revision is available in the remote repository [3b355db864]
Launching `https://github.com/nextflow-io/hello` [deadly_goodall] DSL1 - revision: baba3959d7 [v1.1]
WARN: The use of `echo` method has been deprecated
executor >  local (4)
[da/c897a0] process > sayHello (3) [100%] 4 of 4 ✔
Bojour world! (version 1.1)

Ciao world! (version 1.1)

Hola world! (version 1.1)

Hello world! (version 1.1)
```

**Comment by DininduSenanayake** [comment]
Or not, it is triggering the same error with `Java/17` as well 

```bash
$ NXF_VER=22.10.0 nextflow run nextflow-io/hello -r v1.1
N E X T F L O W  ~  version 22.10.0
NOTE: Your local project version looks outdated - a different revision is available in the remote repository [3b355db864]
Launching `https://github.com/nextflow-io/hello` [grave_crick] DSL1 - revision: baba3959d7 [v1.1]
BUG! exception in phase 'semantic analysis' in source unit 'Script_085f23c1' Unsupported class file major version 64


(nf-core) 11:42:35 wbn004 ~/session1 $ ml | grep "Java"
 16) GLib/2.64.0-GCCcore-9.2.0              33) PCRE2/10.35-GCCcore-9.2.0              50) Java/1
```

**Comment by DininduSenanayake** [comment]
AH..I have figured it out. 

Below is the content of `/nesi/project/nesi02659/nextflow-workshop/init-nf-day1`

```bash
#!/bin/bash

module purge  >/dev/null 2>&1
module load Miniconda3
source $(conda info --base)/etc/profile.d/conda.sh
export CONDA_ENVS_PATH=/nesi/project/nesi02659/.conda/envs

module load Singularity/3.11.3
module load Graphviz/2.42.2-GCC-9.2.0
module load Java/17 

conda activate nf-core

printf "Nexflow-workshop DAY 1 environment is ready \u2705\n"
```
* Looks like the order matters here. as loading the Java module first and then activating the conda environment over-rides 17 with 20
```bash
$ module load Singularity/3.11.3
$ module load Graphviz/2.42.2-GCC-9.2.0
$ module load Java/17 
$ conda activate nf-core
$ java -version
openjdk version "20-internal" 2023-03-21
OpenJDK Runtime Environment (build 20-internal-adhoc..src)
OpenJDK 64-Bit Server VM (build 20-internal-adhoc..src, mixed mode, sharing)
```
Switch it around

```
$ module load Java/17

$ java -version
java version "17.0.1" 2021-10-19 LTS

 NXF_VER=22.10.0 nextflow run nextflow-io/hello -r v1.1
N E X T F L O W  ~  version 22.10.0
NOTE: Your local project version looks outdated - a different revision is available in the remote repository [3b355db864]
Launching `https://github.com/nextflow-io/hello` [jovial_cuvier] DSL1 - revision: baba3959d7 [v1.1]
WARN: The use of `echo` method has been deprecated
executor >  local (4)
[f9/eaeba0] process > sayHello (3) [100%] 4 of 4 ✔
Ciao world! (version 1.1)

Bojour world! (version 1.1)

Hola world! (version 1.1)

Hello world! (version 1.1)
```
